api/controllers/game/ready.js

```diff
@@ -1,55 +1,52 @@
-const gameAPI = sails.hooks['customgamehook'];
 const userAPI = sails.hooks['customuserhook'];
 
-module.exports = function (req, res) {
-  if (req.session.game && req.session.usr) {
-    const promiseGame = gameAPI.findGame(req.session.game);
-    const promiseUser = userAPI.findUser(req.session.usr);
-    Promise.all([promiseGame, promiseUser])
-      // Assign player readiness
-      .then(function foundRecords(values) {
-        const [game, user] = values;
-        let { pNum } = user;
-        let bothReady = false;
-        const gameUpdates = {};
-        switch (pNum) {
-          case 0:
-            gameUpdates.p0Ready = !game.p0Ready;
-            if (game.p1Ready) {
-              bothReady = true;
-            }
-            break;
-          case 1:
-            gameUpdates.p1Ready = !game.p1Ready;
-            if (game.p0Ready) {
-              bothReady = true;
-            }
-            break;
+module.exports = async function (req, res) {
+  try {
+    const game =  await sails.helpers.lockGame(req.session.game);
```

Comment:
I refactored this to use `async/await` largely so that this line is easy to use and work with without needing an extra layer of `.then()` to pass the game through the promise chain. Now consumers can lock a game and query for it by simply `awaiting` the `lockGame()` helper